### PR TITLE
Opj 4 5 add data loader

### PR DESCRIPTION
**Pull request by slevertbiot**

### CODE REVIEW

**Review by mspratt-biot — APPROVED** [review]

File: udmif/api/src/site/SiteDataSource.ts [hunk 1]
```diff
@@ -34,16 +38,27 @@ export class SiteDataSource extends GraphQLDataSource {
...
   async getSiteDeviceStatus(name: string, deviceName: string): Promise<Status> {
     const devices = (await this.getSiteValidation(name)).devices;
     return devices ? devices[deviceName]?.status : null;
   }
+
+  private async batchLoadSitesById(names: readonly string[]): Promise<Site[]> {
+    const returnedSites = await this.siteDAO.getAllIn('name', names);
+
+    return await Promise.all(
+      names.map(async (name) => {
```
Comment (neil-buildingsiot) [inline udmif/api/src/site/SiteDataSource.ts:58]: do we need async here?

File: udmif/api/src/site/SiteDataSource.ts [hunk 2]
```diff
@@ -34,16 +38,27 @@ export class SiteDataSource extends GraphQLDataSource {
   }
 
   async getSite(name: string): Promise<Site> {
-    return this.siteDAO.getOne({ name });
+    return this.sitesDataLoader.load(name);
   }
 
   async getSiteValidation(name: string): Promise<SiteValidation> {
-    const site = await this.siteDAO.getOne({ name });
+    const site = await this.getSite(name);
```
Comment (neil-buildingsiot) [inline udmif/api/src/site/SiteDataSource.ts:45]: If we are just doing validation for a single site, how does the performance of using the dataloader (which does an 'in' query) compare to doing a regular lookup?

Comment (slevertbiot) [reply udmif/api/src/site/SiteDataSource.ts:45]: Feels the same.

File: udmif/api/src/site/SiteDataSource.ts (hunk 1 above)
Comment (slevertbiot) [reply udmif/api/src/site/SiteDataSource.ts:58]: will fix.

**Review by neil-buildingsiot — APPROVED** [review]

**Pull request merged**